src/browser/LiveUpdatePlugin.js

```diff
@@ -1,9 +1,169 @@
+    logger.trace('trace',"__sendConfigRequest: id = " + id + ", url = " + url + ", params = " + JSON.stringify(params) );
+    
+    if (!__isEmpty(params)) {
+      for (var key in params) {
+        if (params.hasOwnProperty(key)) {
+          configurationServiceRequest.setQueryParameter(key, params[key]);
+        }
+      }
+    }
+    configurationServiceRequest.send().then(
+      (response) => {
+        var json = response.responseJSON;
+        if (typeof json === "undefined") {
+          logger.fatal('fatal', "__sendConfigRequest: invalid JSON response");
+          json = {};
+        }
+        var data = json["data"];
+        data["expiresAt"] = json["expiresAt"];
+        let configuration = new __configurationInstance(id, data);
+        WLLiveupdateCache.saveConfiguration(id, configuration);
+        resolve(configuration);
+      },
+      (error) => {
+        logger.fatal('fatal', "__sendConfigRequest: error while retriving configuration from server. error = " + JSON.stringify(error));
+        reject(error);
+      }
+    );
+  });
+}
+
+function __obtainConfiguration(id, url, params, useCache, success, error) {
+  if (WLLiveupdateCache.getConfiguration(id) && useCache) {
+    const cachedConfig = WLLiveupdateCache.getConfiguration(id);
+    logger.debug('debug', "__obtainConfiguration: Retrieved cached configuration. configuration = " + JSON.stringify(cachedConfig));
+    success(cachedConfig);
+  } else {
+    __sendConfigRequest(id, url, params).then(
+      (configuration) => {
+        success(configuration);
+        logger.debug('debug', "__obtainConfiguration: Retrieving new configuration from server. configuration = " + JSON.stringify(configuration));
+      }, (er) => {
+        error(er);
+        logger.debug('debug', "__obtainConfiguration: Error in Retrieving configuration from server. Error = " + JSON.stringify(er));
+      });
+  }
+}
+
+function __getConfigurationWithSegmentId(segmentId, useClientCache, success, error) {
+  const encodedSegment = encodeURI(segmentId);;
+  const url = serviceURL + "/" + encodedSegment;
+  logger.debug('debug', "__obtainConfiguration: segment = " + segmentId + ", useCache = " + useClientCache + ", url = " + url);
+  __obtainConfiguration(segmentId, url, {}, useClientCache, success, error);
+}
+
+function __getConfigurationWithParams(params, useClientCache, success, error) {
+  const url = serviceURL;
+  const id = __buildIDFromParams(params)
+  logger.debug('debug', "__obtainConfiguration: params = " + JSON.stringify(params) + ", useCache = " + useClientCache + ", url = " + url);
+  __obtainConfiguration(id, url, params, useClientCache, success, error);
+}
+
+function getConfiguration(success, error, options) {
+  logger = WL.Logger.create({pkg: 'com.mfp.liveupdate'});
+  if (typeof options[0] !== "undefined") {
+    options = options[0];
+  }
+  const segmentId = (typeof options['segmentId'] !== "undefined") ? options['segmentId'] : 'all';
```

Comment:
Shouldn't it be `options[segmentId]`  ?

Reply:
Nope, We are extracting `segmentId` based on the key value.